### PR TITLE
Throw NotImplementedException in std instead of strings

### PR DESCRIPTION
**Pull request by RealyUniqueName**
Any opinions?

### CODE REVIEW

File: std/cs/internal/HxObject.hx [hunk 1]
```diff
@@ -276,7 +276,7 @@ class HxEnum {
 	}
 
 	public function getTag():String {
-		return throw 'Not Implemented';
+		return throw new haxe.exceptions.NotImplementedException('Not Implemented');
```
Comment (Gama11) [inline std/cs/internal/HxObject.hx]: Why specify the message in these cases, seems you made it optional with "Not Implemented" as the default?

Comment (RealyUniqueName) [reply std/cs/internal/HxObject.hx]: Indeed. Updated.

File: std/cs/internal/HxObject.hx [hunk 2]
```diff
@@ -276,7 +276,7 @@ class HxEnum {
 	}
 
 	public function getTag():String {
-		return throw 'Not Implemented';
+		return throw throw new haxe.exceptions.NotImplementedException();
```
Comment (Gama11) [inline std/cs/internal/HxObject.hx]: Failed find-and-replace? :)

Comment (RealyUniqueName) [reply std/cs/internal/HxObject.hx]: yep

**Pull request merged**